File: solutions/zig/scrabble-score/1/scrabble_score.zig
const std = @import("std");

pub fn score(s: []const u8) u32 {
    var total: u8 = 0;

    for (s) |c| {
        switch (std.ascii.toLower(c)) {
            'a', 'e', 'i', 'o', 'u', 'l', 'n', 'r', 's', 't' => total += 1,
            'd', 'g' => total += 2,
            'b', 'c', 'm', 'p' => total += 3,
            'f', 'h', 'v', 'w', 'y' => total += 4,
            'k' => total += 5,
            'j', 'x' => total += 8,
            'q', 'z' => total += 10,
            else => {},
        }
    }

    return total;
}

File: solutions/go/scrabble-score/1/scrabble_score.go
package scrabblescore

import "strings"

func Score(word string) int {
	var total int
	upper := strings.ToUpper(word)
	for _, r := range upper {
		switch r {
		case 'A', 'E', 'I', 'O', 'U', 'L', 'N', 'R', 'S', 'T':
			total++
		case 'D', 'G':
			total += 2
		case 'B', 'C', 'M', 'P':
			total += 3
		case 'F', 'H', 'V', 'W', 'Y':
			total += 4
		case 'K':
			total += 5
		case 'J', 'X':
			total += 8
		case 'Q', 'Z':
			total += 10
		default:
		}
	}
	return total
}
